File: c_sharp/Enumeration/Enumeration/GeneralStatus.cs
using PeerageHRIS.Enumeration;
using PeerageHRIS.Resources.Models.Training;
using System.ComponentModel.DataAnnotations;

namespace PeerageHRIS.Enumeration
{
    public enum GeneralStatus
    {
        [Display(Name = "Pending", ResourceType = typeof(TrainingResource))]
        Pending,
        [Display(Name = "Submitted", ResourceType = typeof(TrainingResource))]
        Submitted,
        [Display(Name = "Approved", ResourceType = typeof(TrainingResource))]
        Approved,
        [Display(Name = "Rejected", ResourceType = typeof(TrainingResource))]
        Rejected,
        [Display(Name = "Completed", ResourceType = typeof(TrainingResource))]
        Completed,
        [Display(Name = "Canceled", ResourceType = typeof(TrainingResource))]
        Canceled
    }
} 

File: lib/core/enums/general_status.dart
/// Represents the general status for workflows, approvals, or processes in the system.
///
library;

import 'package:freezed_annotation/freezed_annotation.dart';

part 'general_status.g.dart';

@JsonEnum(alwaysCreate: true)
enum GeneralStatus {
  @JsonValue(0)
  pending,

  @JsonValue(1)
  submitted,

  @JsonValue(2)
  approved,

  @JsonValue(3)
  rejected,

  @JsonValue(4)
  completed,

  @JsonValue(5)
  canceled,
}
